roles/StackStorm.nodejs/tasks/nodejs_yum.yml

```diff
@@ -10,6 +10,24 @@
   until: _task is succeeded
   tags: nodejs
 
+# nodesource updates the baseurl (which includes the major nodejs version)
+# in their yum repo release rpm without increasing the rpm's version number.
+# So we must manually remove the repo to get yum to add the correct release.
+- name: Determine if nodesource repo upgrade is required (ie when updating nodejs major version)
```

Comment:
dumb idea, but i'll through it out there...

I see that this ansible role currently relies on the RPM installation for the nodesource repo.

the way the the Puppet module does this is basically create a template file and write the "desired state" of the nodejs repo into that file, updating the URLs, etc


if not, then this code looks OK to me

Reply:
FYI here are some links to the puppet code:
https://github.com/voxpupuli/puppet-nodejs/blob/master/manifests/repo/nodesource/yum.pp#L21-L33

those parameters/variables come from:
https://github.com/voxpupuli/puppet-nodejs/blob/master/manifests/repo/nodesource.pp#L43

Reply:
talking to myself here, we probably want to keep the RPM version instead of the template because that's how it's done now and in the Ubuntu version we install a .deb in the same way

Reply:
templating the repo file would avoid hacking around nodesource's awkward RPM. I'm fine with either approach.

If we switch to templating, then should we might want a task that removes the node.js repo rpm so that (a) there is only one "blessed" way to install the repo file, and (b) the playbook does not mess up existing installations by editing an yum-managed file.

Reply:
actually.... looking at the Ubuntu version it uses the `apt_repostiory` ansible module

looks like there is a similar module for `yum_repository` https://docs.ansible.com/ansible/latest/modules/yum_repository_module.html

this would make the behavior consistent across OSes to use `yum_repository`

Reply:
https://github.com/StackStorm/ansible-st2/blob/52ed353c8554b5819f7d587fd32858f348c6aec2/roles/StackStorm.nodejs/tasks/nodejs_apt.yml#L26-L27

As you pointed out, the repo deb file already includes the version number, so it should upgrade easily already.

docs on how to do this (step 3 is the node repo for deb)
https://github.com/nodesource/distributions#manual-installation

I like the idea of using the `yum_repository` module to avoid these shenanigans.